src/pages/api/modules.ts

```diff
@@ -0,0 +1,31 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import data from "../../../data.json";
+
+const modules = (req: NextApiRequest, res: NextApiResponse) => {
+  const searchTerm = req.query.search;
+
+  if (!searchTerm) {
+    res.statusCode = 200;
+    return res.json(data);
```

Comment:
I suppose I'm fine with sending all the data, but then we might as well cache it in the client side and then we don't have to query on keypress.

Once the data gets larger, we should implement have `!searchTerm` return the first 20 modules or something.

Reply:
Imma file this one under "Most likely not production ready but more of a discussion topic." 😛